openpype/plugins/publish/collect_colorspace_data.py

```diff
@@ -24,10 +24,11 @@ class ExtractColorspaceData(publish.Extractor,
             }
         }
     """
-    label = "Extract Colorspace data"
-    order = pyblish.api.ExtractorOrder + 0.49
+    label = "Collect Colorspace data"
+    order = pyblish.api.CollectorOrder + 0.49
```

Comment:
Pretty sure this will break quite a few cases - in the majority of cases representations are NOT set during Collecting stage yet but get defined by the Extractors. For all those cases this would now basically not trigger this logic. Seems like a huge change, no? Unless we're changing all those extractors that are expecting it?

Reply:
I wonder if you had to do this for the same reason I ended up having to add this line:
```python
instance.data["colorspace"] = colorspace
```
on the existing "openpype\hosts\traypublisher\plugins\publish\collect_explicit_colorspace.py" plugin that @BigRoy created. It would be better if you could instead reuse that plugin and make it work to your needs.

I basically had to add that line because at the point where the `collect_explicit_colorspace` runs the representations that get submitted to the farm hadn't been generated but adding it to the instance data is enough for then setting the right colorspace when they do